dspace-api/src/main/java/org/dspace/content/service/EntityService.java

```diff
@@ -0,0 +1,135 @@
+     *                  that is equal to the label param
+     * @throws SQLException If something goes wrong
+     */
+    List<Relationship> getRelationsByLabel(Context context, String label) throws SQLException;
+
+    /**
+     * Retrieves the list of relationships that have a relationshiptype that contains the EntityType for the given
+     * Entity
+     * in either the leftEntityType or the rightEntityType variables
+     * @param context   The relevant DSpace context
+     * @param entity    The Entity for which the EntityType should be checked for relationships
+     * @return The list of relationships that each contain a relationshiptype in which there is a right or left
+     * entity type that
+     *                  is equal to the entity type for the given entity
+     * @throws SQLException If something goes wrong
+     */
+    List<RelationshipType> getAllRelationshipTypes(Context context, Entity entity) throws SQLException;
+
+    /**
+     * Retrieves the list of relationships that have a relationshiptype that contains the EntityType for the given
+     * Entity
+     * in the leftEntityType
+     * @param context   The relevant DSpace context
+     * @param entity    The Entity for which the EntityType should be checked for relationships
+     * @return The list of relationships that each contain a relationshiptype in which there is a left entity type that
+     *                  is equal to the entity type for the given entity
+     * @throws SQLException If something goes wrong
+     */
+    List<RelationshipType> getLeftRelationshipTypes(Context context, Entity entity) throws SQLException;
+
+    /**
+     * Retrieves the list of relationships that have a relationshiptype that contains the EntityType for the given
+     * Entity
+     * in the rightEntityType
+     * @param context   The relevant DSpace context
+     * @param entity    The Entity for which the EntityType should be checked for relationships
+     * @return The list of relationships that each contain a relationshiptype in which there is a right entity type that
+     *                  is equal to the entity type for the given entity
+     * @throws SQLException If something goes wrong
+     */
+    List<RelationshipType> getRightRelationshipTypes(Context context, Entity entity) throws SQLException;
+
+    /**
+     * Retrieves a list of RelationshipType objects for which either their left or right label is equal to the
+     * label parameter that's being passed along
+     * @param context   The relevant DSpace context
+     * @param label     The label for which the relationshiptype's labels must be checked
+     * @return The list of relationshiptypes that each contain a left or right label that is equal
+     *                  to the given label parameter
+     * @throws SQLException If something goes wrong
+     */
+    List<RelationshipType> getRelationshipTypesByLabel(Context context, String label) throws SQLException;
```

Comment:
a paginated method should be provided

Reply:
This has been included in https://jira.duraspace.org/browse/DS-4224